libs/providers/multi-provider/src/lib/strategies/ComparisonStrategy.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  BaseEvaluationStrategy,
+  FinalResult,
+  ProviderResolutionResult,
+  StrategyPerProviderContext,
+} from './BaseEvaluationStrategy';
+import { EvaluationContext, FlagValue, Provider } from '@openfeature/server-sdk';
+
+/**
+ * Evaluate all providers in parallel and compare the results.
+ * If the values agree, return the value
+ * If the values disagree, return the value from the configured "fallback provider" and execute the "onMismatch"
+ * callback if defined
+ */
+export class ComparisonStrategy extends BaseEvaluationStrategy {
+  override runMode = 'parallel' as const;
+
+  constructor(
+    private fallbackProvider: Provider,
+    private onMismatch?: (resolutions: ProviderResolutionResult<FlagValue>[]) => void,
+  ) {
+    super();
+  }
+
+  override determineFinalResult<T extends FlagValue>(
+    strategyContext: StrategyPerProviderContext,
+    context: EvaluationContext,
+    resolutions: ProviderResolutionResult<T>[],
+  ): FinalResult<T> {
+    let value: T | undefined;
+    let fallbackResolution: ProviderResolutionResult<T> | undefined;
+    let mismatch = false;
+    for (const resolution of resolutions) {
+      if (resolution.thrownError || resolution.details?.errorCode) {
+        return this.collectProviderErrors(resolutions);
+      }
+      if (resolution.provider === this.fallbackProvider) {
+        fallbackResolution = resolution;
+      }
+      if (typeof value !== 'undefined' && value !== resolution.details!.value) {
```

Comment:
We can avoid the non-null assertion if we use types as described in the comment in `BaseEvaluationStrategy.ts`